### PR TITLE
Remove grunt

### LINKED ISSUE #2749: Migrate to gulp and move grunt to a module/tip
Based on discussion and vote [here](https://groups.google.com/forum/m/?utm_medium=email&utm_source=footer#!msg/jhipster-dev/pSQXz_NWcZ0/iqC3tqUHDQAJ)

Make gulp as default front end build tool.
Remove grunt from main generator 
Make grunt build a module or tip
Improve the gulp build where possible
Add gulp notify coz its cool
Reorganize and simplify gulp build
Move utils used to its own file

### PR DESCRIPTION
**Pull request by deepu105**
removed grunt as an option
remove all reference to grunt
fix tests
fix travis
move gulp utils to its own file
improve gulp format
add plumber and gulp notify.

Notify is very cool and handy especially when doing gulp serve, you make a mistake and notification pops up alerting you immediately, llok at below when i did an error in the sass file, i dont have to go to logs to see there is an error, i have immediate notification feedback with a beep

image

fix #2749

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +20,7 @@ env:
   matrix:
     - JHIPSTER=app-from-scratch JHIPSTER_NODE_CACHE=0 RUN_APP=0
     - JHIPSTER=app-gradle RUN_APP=0
-    - JHIPSTER=app-gulp GRUNT=0
+    - JHIPSTER=app-gulp
```
Comment (deepu105) [inline .travis.yml]: @pascalgrimaud this is now same as app-default, so either we can remove this or add something useful instead

File: travis/scripts/01-generate-project.sh [hunk 2]
```diff
@@ -5,13 +5,7 @@ set -ev
 #-------------------------------------------------------------------------------
 mv -f $JHIPSTER_SAMPLES/$JHIPSTER $HOME/
 cd $HOME/$JHIPSTER
-if [ $GRUNT == 1 ]; then
-  rm -Rf $HOME/$JHIPSTER/node_modules/.bin/*gulp*
-  rm -Rf $HOME/$JHIPSTER/node_modules/*gulp*
-else
-  rm -Rf $HOME/$JHIPSTER/node_modules/.bin/*grunt*
-  rm -Rf $HOME/$JHIPSTER/node_modules/*grunt*
```
Comment (pascalgrimaud) [inline travis/scripts/01-generate-project.sh]: keep theses 2 lines for now (delete *grunt* lib), to avoid conflict

File: .travis.yml (hunk 1 above)
Comment (pascalgrimaud) [inline .travis.yml]: Yes we can remove the variable grunt

File: travis/scripts/01-generate-project.sh (hunk 2 above)
Comment (deepu105) [inline travis/scripts/01-generate-project.sh]: you mean these 2 
```
rm -Rf $HOME/$JHIPSTER/node_modules/.bin/*grunt*
rm -Rf $HOME/$JHIPSTER/node_modules/*grunt*
```

Comment (pascalgrimaud) [inline travis/scripts/01-generate-project.sh]: Yes exact! There was an issue in the past, about using cache for node_modules, I don't remember exactly. So to secure this PR, until I have time to clean https://github.com/jhipster/jhipster-travis-build with grunt lib

**Comment by deepu105** [comment]
can someone test the notification feature in OSX and Linux?
Just do gulp serve and make an error in sass file or something and it should pop up

File: travis/scripts/01-generate-project.sh (hunk 2 above)
Comment (deepu105) [inline travis/scripts/01-generate-project.sh]: ok done

**Comment by sdoxsee** [comment]
I'm all for removing options for simplicity...but what is usage like for grunt? I've always used it. Perhaps make gulp the yo default for a while first before removing?

**Comment by deepu105** [comment]
@sdoxsee we are not ditching grunt altogether, we will make a module which will add grunt support so that if anyone wants to use grunt they still can. In the meanwhile we want to simplify the main generator and offering two solutions for the same thing when one of it is clearly a better choice seems counter productive keeping only gulp in main generator means we can provide it better support and its easy for maintenance. We offer maven and gradle but that atleast makes more sense given both are different and there is clearly following for both regardless of publicity where as any front end guy who has used gulp will unlikely go back to grunt

**Comment by sdoxsee** [comment]
Thumbs up then! :+1:

**Comment by cicoub13** [comment]
Notification works on OSX (only with sass) :+1:

sass-notification

**Comment by deepu105** [comment]
It should work on any error, did you had any console error for which there
was no notification?

**Comment by deepu105** [comment]
@jdubois can you merge this if you are ok? its giving conflicts everyday

**Comment by jdubois** [comment]
Sorry I have no time to review it now... If it's just removing Grunt, as we all agree, just do it.

**Comment by cicoub13** [comment]
Yes, it works for any console error. It would be awesome if js changes trigger jshint task with error notification (like this https://gist.github.com/rudijs/9148283 ?).

**Comment by deepu105** [comment]
@cicoub13 yes its absolutely possible, if you want you can do a PR for that

**Comment by cicoub13** [comment]
OK, i'll do it.

**Pull request merged**